l10n_us_form_1099/models/res_partner.py

```diff
@@ -0,0 +1,22 @@
+# Copyright 2017 Open Source Integrators <https://www.opensourceintegrators.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class ResPartner(models.Model):
+    _inherit = "res.partner"
+
+    is_1099 = fields.Boolean('Is a 1099')
+
+    @api.onchange('is_1099')
+    def _on_change_is_1099(self):
+
+        if self.is_1099 and not self.supplier:
+            self.supplier = True
+
+    @api.onchange('supplier')
+    def _on_change_supplier(self):
+
+        if self.is_1099 and not self.supplier:
```

Comment:
I think it may make sense to also add a constraint to not allow a 1099 on a non-supplier?